cmd/arc/services/callbacker.go

```diff
@@ -20,23 +23,16 @@ func StartCallbacker(logger *slog.Logger, appConfig *config.ArcConfig) (func(),
 
 	config := appConfig.Callbacker
 
-	callbackSender, err := callbacker.NewSender(&http.Client{Timeout: 5 * time.Second}, logger)
-	if err != nil {
-		return nil, fmt.Errorf("callbacker failed: %v", err)
-	}
+	var (
+		store      store.CallbackerStore
+		sender     *callbacker.CallbackSender
+		dispatcher *callbacker.CallbackDispatcher
 
-	callbackDispatcher := callbacker.NewCallbackDispatcher(callbackSender, config.Pause)
+		server       *callbacker.Server
+		healthServer *grpc.Server
 
-	server := callbacker.NewServer(callbackDispatcher, callbacker.WithLogger(logger.With(slog.String("module", "server"))))
-	err = server.Serve(config.ListenAddr, appConfig.GrpcMessageSize, appConfig.PrometheusEndpoint)
-	if err != nil {
-		return nil, fmt.Errorf("GRPCServer failed: %v", err)
-	}
-
-	healthServer, err := StartHealthServerCallbacker(server, config.Health, logger)
-	if err != nil {
-		return nil, fmt.Errorf("failed to start health server: %v", err)
-	}
+		err error
+	)
```

Comment:
If you'd move the creation of `stopFn` to the end of this function, you wouldn't have to declare these variables here and then check if they are != null in the `stopFn`

Reply:
I missed that, thanks for pointing it out. Nice!